### PR TITLE
Fix github issue #1265

### PR DESCRIPTION
**Pull request by agran2018**
**Context:** Fix github issue [#1265].  This PR ensures the type consistency of the qubit Hamiltonian terms constructed by the 'convert_observable' function from an OpenFermion QubitOperator with respect to the same observable built directly using PennyLane operations.

**Description of the Change:**
The fix was implemented in the ``_qubit_operator_to_terms`` function of the ``qchem/structure`` module. An extra unit test was added to ``test_convert_observable.py``

**Benefits:**
Properties of Hamiltonian consistently return same type.

**Possible Drawbacks:**
None

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/1265

### CODE REVIEW

File: qchem/CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,18 @@
 
 <h3>Bug fixes</h3>
 
+* The types of the Hamiltonian terms built from an OpenFermion ``QubitOperator`` using the
+  ``convert_observable`` function, are the same with respect to the analogous observable
+  built directly using PennyLane operations.
```
Comment (josh146) [inline qchem/CHANGELOG.md:11]:
```suggestion
* The types of the Hamiltonian terms built from an OpenFermion ``QubitOperator`` using the
  ``convert_observable`` function, are the same with respect to the analogous observable
  built directly using PennyLane operations.
  [(#1523)](https://github.com/PennyLaneAI/pennylane/pull/1523)
```

**Review by josh146 — COMMENTED** [review]
@agran2018 I'm not familiar with this bug, so haven't reviewed it 🙂 However, I do have a question: since this is a bugfix, should this PR be merged into the release candidate branch?

(@antalszava)

**Comment by antalszava** [comment]
Thanks for the heads-up here Josh :)

Indeed, this should be incorporated in the v0.17.0 release (provided it's merged by Monday). The only catch is, that this branch already has changes incorporated from `master`. We would like to, however, add this update to the https://github.com/PennyLaneAI/pennylane/tree/v0.17.0-rc0 branch.

There may be two ways to do this:

1. The changes from `master` are undone in this PR (this is quite tedious with having merge commits too).
2. The PR is recreated by branching off from `v0.17.0-rc0 branch` locally and rebasing to point to that branch.

The base branch can be changed by clicking the "Edit" button at the top right of the PR:
Screenshot from 2021-08-13 08-23-06

File: qchem/pennylane_qchem/qchem/structure.py [hunk 2]
```diff
@@ -573,8 +573,12 @@ def _qubit_operator_to_terms(qubit_operator, wires=None):
             (
                 coef,
                 qml.operation.Tensor(*[xyz2pauli[q[1]](wires=wires[q[0]]) for q in term])
-                if term
-                else qml.operation.Tensor(qml.Identity(wires[0]))
+                if len(term) > 1
+                else (
+                    xyz2pauli[term[0][1]](wires=wires[term[0][0]])
+                    if len(term) == 1
```
Comment (rmoyard) [inline qchem/pennylane_qchem/qchem/structure.py:579]: Is the check necessary can `len(term)`be equal to 0?

**Review by rmoyard — COMMENTED** [review]
Except the one comment I've left, everything is looking good to me 💯  and you should also follow Antal's comment on rebasing the PR with the release candidate `v0.17.0-rc0` branch. Thanks again

File: qchem/pennylane_qchem/qchem/structure.py (hunk 2 above)
Comment (zeyueN) [reply qchem/pennylane_qchem/qchem/structure.py:579]: Yes
image

**Comment by rmoyard** [comment]
New PR was created https://github.com/PennyLaneAI/pennylane/pull/1525

**Pull request closed without merging**